scripts/Crawlers/BaseCrawler.py

```diff
@@ -0,0 +1,209 @@
+from datalad import api
+import git
+import os
+import re
+import requests
+import abc
+import json
+
+
+class BaseCrawler:
+    def __init__(self, github_token, config_path, verbose, force):
+        self.repo = git.Repo()
+        self.username = self._check_requirements()
+        self.github_token = github_token
+        self.config_path = config_path
+        self.verbose = verbose
+        self.force = force
+        self.git = git
+        self.datalad = api
+
+    @abc.abstractmethod
+    def get_all_dataset_metadata(self):
+        """
+        Get relevant datasets' metadata from platform.
+
+        Retrieves datasets' metadata that needs to be in CONP-datasets
+        from platform specific to each crawler like Zenodo, OSF, etc.
+
+        Returns:
+            List of metadata of relevant datasets. Each metadata is a
+            dictionary and requires the key "title" to be associated
+            with the dataset's title. For example:
+
+            [{
+                "title": "PERFORM Dataset Example",
+                "creators: ["Joey", "Daniel", "Bob"]
+             },
+             {
+                "title": "SIMON dataset example",
+                "version": 1.4.2
+             },
+             ...
+            ]
+        """
+        return []
+
+    @abc.abstractmethod
+    def add_new_dataset(self, metadata, dataset_dir):
+        """
+        Configure and add newly created dataset.
+
+        This is where newly crawled dataset which are not present locally are
+        created. This is where a one time configuration such as modifying
+        .gitignore, .gitattribute or no-annex some files can be done.
+
+        Parameter:
+        metadata (dict): Dictionary containing metadata on
```

Comment:
Mention that the metadata (identifier) will be an element of the list of identifiers returned by the previous function.